File: mobile_app/lib/notices/models/notice.dart
import 'package:mobile_app/shared/required_enums.dart';

class Notice {
  final String id;
  final String title;
  final DateTime publishedAt;
  final DateTime? scheduledAt;
  final String description;
  final NoticePriority priority;

  Notice({
    required this.id,
    required this.title,
    required this.publishedAt,
    this.scheduledAt,
    required this.description,
    required this.priority,
  });

  factory Notice.fromMap(Map<String, dynamic> map) {
    return Notice(
      id: map['id'],
      title: map['title'],
      publishedAt: DateTime.parse(map['published_at']).toLocal(),
      scheduledAt: map['scheduled_at'] == null
          ? null
          : DateTime.parse(map['scheduled_at']).toLocal(),
      description: map['description'],
      priority: getPriority(map['priority']),
    );
  }

  static NoticePriority getPriority(String s) {
    if (s == 'urgent') return NoticePriority.urgent;
    if (s == 'important') return NoticePriority.important;

    return NoticePriority.info;
  }
}

File: admin_web/src/models/notice.ts

export class Notice {
  id: string | null;
  title: string;
  publishedAt: Date;
  scheduledAt: Date | null;
  description: string;
  priority: string;

  constructor(
    id: string |null,
    title: string,
    publishedAt: Date,
    scheduledAt: Date | null,
    description: string,
    priority: string
  ) {
    this.id = id;
    this.title = title;
    this.publishedAt = publishedAt;
    this.scheduledAt = scheduledAt;
    this.description = description;
    this.priority = priority;
  }

  // Convert API response to Notice object
  static fromMap(map: any): Notice {
    return new Notice(
      map.id,
      map.title,
      new Date(map.published_at),
      map.scheduled_at ? new Date(map.scheduled_at) : null,
      map.description,
      map.priority
    );
  }

  // Convert Notice object to API 
  toMap() {
    return {
      id: this.id,
      title: this.title,
      published_at: this.publishedAt.toISOString(),
      scheduled_at: this.scheduledAt
        ? this.scheduledAt.toISOString()
        : null,
      description: this.description,
      priority: this.priority,
    };
  }
}